### PR TITLE
XDPForwardingTest: Replaced sender program with DPDK to get more PPS.

### PR DESCRIPTION
**Pull request by abhimarathe**
Updated sender program with DPDK to generate more PPS for the test.

```
[LISAv2 Test Results Summary]
Test Run On           : 10/27/2020 22:10:19
ARM Image Under Test  : Canonical : 0001-com-ubuntu-server-focal : 20_04-lts : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:11

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              PERF-XDP-FORWARD-COMPARISON                                                       PASS                 9.68 
      ARMImageName: Canonical 0001-com-ubuntu-server-focal 20_04-lts latest, Networking: SRIOV, OverrideVMSize: Standard_D64_v3, TestLocation: eastus2, Kernel Version: 5.4.0-1031-azure


Logs can be found at C:\LISAv2\TX87\TestResults\2020-10-27-15-09-29-4822
```

### CODE REVIEW

File: Testscripts/Windows/PERF-XDP-TX-FORWARD.ps1 [hunk 1]
```diff
@@ -43,18 +43,18 @@ function Main {
         # Add modes according to testcase
         if ( $CurrentTestData.testName -match "TX" ){
             Add-Content -Value "modes=(xdp)" -Path $constantsFile
-            $packetCount=10000000
+            $testDuration=15
```
Comment (juhlee-microsoft) [inline Testscripts/Windows/PERF-XDP-TX-FORWARD.ps1:46]: This change is from measuring packet counts to testing time. Do you buy this change with DPDK folks?

File: Testscripts/Linux/XDPForwardingTest.sh [hunk 2]
```diff
@@ -152,16 +179,29 @@ for mode in ${modes[@]}; do
...
 		LogMsg "${forwarder} vm setup completed for ip tables"
 	fi
 
 	# Start pktgen on Sender
 	forwarderSecondMAC=$((ssh ${forwarder} "ip link show ${nicName}") | grep ether | awk '{print $2}')
-	LogMsg "Starting pktgen on ${sender}"
-	start_pktgen ${sender} ${cores} ${pktgenDir} ${nicName} ${forwarderSecondMAC} ${forwarderSecondIP} ${packetCount}
-	sleep 5
+	LogMsg "Starting dpdk-testpmd on ${sender}"
+	#start_pktgen ${sender} ${cores} ${pktgenDir} ${nicName} ${forwarderSecondMAC} ${forwarderSecondIP} ${packetCount}
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/XDPForwardingTest.sh]: Don't leave the code you are not going to use.

File: Testscripts/Linux/XDPForwardingTest.sh [hunk 3]
```diff
@@ -152,16 +179,29 @@ for mode in ${modes[@]}; do
...
+	vfName=$(get_vf_name ${nicName})
+	sender_busaddr=$(ethtool -i ${vfName} | grep bus-info | awk '{print $2}')
+
+	sender_testfwd_cmd="$(Create_Timed_Testpmd_Cmd "${test_duration}" "${core}" "${sender_busaddr}" "${nicName}" txonly "failsafe" "${trx_rx_ips}")"
+	LogMsg "${sender_testfwd_cmd}"
+	testpmd_filename="${LIS_HOME}/dpdk-testfwd-sender-${core}-core-$(date +"%m%d%Y-%H%M%S").log"
+	eval "${sender_testfwd_cmd} 2>&1 > ${testpmd_filename} &"
+	sleep "${test_duration}"
+
+	# kill dpdk-testpmd
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/XDPForwardingTest.sh]: Delete this "kill" command in the comment

File: Testscripts/Windows/PERF-XDP-TX-FORWARD.ps1 (hunk 1 above)
Comment (abhimarathe) [reply Testscripts/Windows/PERF-XDP-TX-FORWARD.ps1:46]: I kept time of 15 seconds coz at the rate of min 10MPPS we will get more than 150 Million packets sent. That's enough for testing packet forwarding.

File: Testscripts/Linux/XDPForwardingTest.sh (hunk 2 above)
Comment (abhimarathe) [reply Testscripts/Linux/XDPForwardingTest.sh]: done!

File: Testscripts/Linux/XDPForwardingTest.sh (hunk 3 above)
Comment (abhimarathe) [reply Testscripts/Linux/XDPForwardingTest.sh]: done!

**Review by juhlee-microsoft — APPROVED** [review]

**Pull request merged**